tests/snappy_pipeline/workflows/test_workflows_cbioportal_export.py

```diff
@@ -0,0 +1,599 @@
+    assert actual == expected
+
+
+def test_cbioportal_vcf2maf_step_part_get_resource_usage(cbioportal_export_workflow):
+    """Tests cbioportalVcf2MafStepPart.get_resource_usage()"""
+    # Define expected: default defined workflow.abstract
+    expected_dict = {"threads": 4, "time": "12:00:00", "memory": "5120M", "partition": None}
+    # Evaluate
+    for resource, expected in expected_dict.items():
+        msg_error = f"Assertion error for resource '{resource}'."
+        actual = cbioportal_export_workflow.get_resource("cbioportal_vcf2maf", "run", resource)
+        assert actual == expected, msg_error
+
+
+# Tests for cbioportalMafStepPart   ----------------------------------------------------------------
+
+
+def test_cbioportal_maf_step_part_get_input_files(cbioportal_export_workflow):
+    """Tests cbioportalMafStepPart.get_input_files()"""
+    base_name = (
+        "work/maf/bwa.mutect.jannovar_annotate_somatic_vcf.dkfz_bias_filter.eb_filter."
+        "P00{i}-T{t}-DNA1-WGS1.dkfz_only.genome_wide/out/"
+        "bwa.mutect.jannovar_annotate_somatic_vcf.dkfz_bias_filter.eb_filter."
+        "P00{i}-T{t}-DNA1-WGS1.dkfz_only.genome_wide.maf"
+    )
+    expected = [base_name.format(i=i, t=t) for i, t in ((1, 1), (2, 1), (2, 2))]
+    actual = cbioportal_export_workflow.get_input_files("cbioportal_maf", "run")
+    assert actual == expected
+
+
+def test_cbioportal_maf_step_part_get_output_files(cbioportal_export_workflow):
+    """Tests cbioportalMafStepPart.get_output_files()"""
+    # Method not implemented
+    with pytest.raises(NotImplementedError):
+        cbioportal_export_workflow.get_output_files("cbioportal_maf", "run")
+
+
+def test_cbioportal_maf_step_part_get_log_file(cbioportal_export_workflow):
+    """Tests cbioportalMafStepPart.get_log_file()"""
+    # Method not implemented
+    with pytest.raises(NotImplementedError):
+        cbioportal_export_workflow.get_log_file("cbioportal_maf", "run")
+
+
+def test_cbioportal_maf_step_part_get_resource_usage(cbioportal_export_workflow):
+    """Tests cbioportalMafStepPart.get_resource_usage()"""
+    # Define expected: default defined workflow.abstract
+    expected_dict = {"threads": 1, "time": "01:00:00", "memory": "2G", "partition": None}
+    # Evaluate
+    for resource, expected in expected_dict.items():
+        msg_error = f"Assertion error for resource '{resource}'."
+        actual = cbioportal_export_workflow.get_resource("cbioportal_maf", "run", resource)
+        assert actual == expected, msg_error
+
+
+# Tests for cbioportalCnaFilesStepPart   -----------------------------------------------------------
+
+
+def test_cbioportal_cna_data_step_part_get_input_files_log2(cbioportal_export_workflow):
+    """Tests cbioportalCnaFilesStepPart.get_input_files() - action 'log2'"""
+    # Define expected
+    base_name = (
+        "SOM_CNV_CALLING/output/bwa.copywriter.P00{i}-T{t}-DNA1-WGS{g}/out/"
```

Comment:
Same here. Additional problem: the somatic CNV calling is split into `somatic_target_seq_cnv_calling` & `somatic_wgs_cnv_calling`. So it is difficult to know from the config only which one has been used (or perhaps both of them).

Reply:
I agree, it is a fairly hidden association and it took me a while to figure out in the code. If config has certain tools it assumes targeted, otherwise WGS.

For now, I simply set a generic 'SOM_CNV_CALLING' step to avoid dealing with it. We can set this as a point for change during a potential refactoring.

```
        if self.config["path_copy_number_step"]:
            if self.config["cnv_tool"] in ["cnvetti_on_target_postprocess", "copywriter"]:
                self.register_sub_workflow(
                    "somatic_targeted_seq_cnv_calling",
                    workdir=self.config["path_copy_number_step"],
                    sub_workflow_name="copy_number_step",
                )
            else:
                self.register_sub_workflow(
                    "somatic_wgs_cnv_calling",
                    workdir=self.config["path_copy_number_step"],
                    sub_workflow_name="copy_number_step",
                )

```